pkg/migration/registry.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Upbound Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	errFmtNewObject          = "failed to instantiate a new runtime.Object using scheme.Scheme for: %s"
+	errFmtNotManagedResource = "specified GVK does not belong to a managed resource: %s"
+)
+
+var (
+	// the default Converter registry
+	registry Registry = make(map[schema.GroupVersionKind]Converter)
```

Comment:
nit: I'd lean towards leaving it to the consumer of this package to initialize and maintain their `Registry` object instead of us providing a global map unless we have to.

Reply:
Makes sense. Planning to address in the next iteration.